contrib/go-redis/redis/redis_test.go

```diff
@@ -313,3 +313,46 @@ func TestAnalyticsSettings(t *testing.T) {
 		assertRate(t, mt, 0.0, WithAnalyticsRate(0.0))
 	})
 }
+
+type testKey string
+
+func TestWithContext(t *testing.T) {
+	opts := &redis.Options{Addr: "127.0.0.1:6379"}
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	client := NewClient(opts, WithServiceName("my-redis"))
+	s1, ctx := tracer.StartSpanFromContext(context.Background(), "span1.name")
+	client = client.WithContext(ctx)
```

Comment:
There's discrepancy here between naming styles. You have `s1` and `s2` and then `ctx` and `ctx2`, `client` and `client2`. Can we please stick to one consistently? Personally I think the one with `*1` and `*2` looks best.

Reply:
Agreed. I'll go with `*1` and `*2`